core/indexer/elasticsearchDatabase.go

```diff
@@ -0,0 +1,328 @@
+package indexer
+
+import (
+	"bytes"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/ElrondNetwork/elrond-go/core/statistics"
+	"github.com/ElrondNetwork/elrond-go/data"
+	"github.com/ElrondNetwork/elrond-go/data/block"
+	"github.com/ElrondNetwork/elrond-go/hashing"
+	"github.com/ElrondNetwork/elrond-go/marshal"
+	"github.com/elastic/go-elasticsearch/v7"
+	"github.com/elastic/go-elasticsearch/v7/esapi"
+)
+
+// elasticSearchDatabaseArgs is struct that is used to store all parameters that are needed to create a elastic search
+// database object
```

Comment:
maybe add: "it contains business logic built over databaseWriterHandler glue code wrapper"

Reply:
done.